cs3api4lab/api/cs3_file_api.py

```diff
@@ -57,53 +57,52 @@ def mount_point(self):
         """
         time_start = time.time()
-        ref = FileUtils.get_reference(file_id, endpoint)
-        stat_info = self.cs3_api.Stat(request=cs3sp.StatRequest(ref=ref),
-                                      metadata=[('x-access-token', self.auth.authenticate())])
+        
         time_end = time.time()
-        self.log.info('msg="Invoked stat" fileid="%s" elapsedTimems="%.1f"' % (file_id, (time_end - time_start) * 1000))
-
-        if stat_info.status.code == cs3code.CODE_OK:
-            self.log.debug('msg="Stat result" data="%s"' % stat_info)
+        self.log.info('msg="Invoked stat" fileid="%s" elapsedTimems="%.1f"' % (file_path, (time_end - time_start) * 1000))
+        stat = self.storage_api.stat(file_path, endpoint)
+        if stat.status.code == cs3code.CODE_OK:
+            self.log.debug('msg="Stat result" data="%s"' % stat)
             return {
-                'inode': {'storage_id': stat_info.info.id.storage_id,
-                          'opaque_id': stat_info.info.id.opaque_id},
-                'filepath': stat_info.info.path,
-                'userid': stat_info.info.owner.opaque_id,
-                'size': stat_info.info.size,
-                'mtime': stat_info.info.mtime.seconds,
-                'type': stat_info.info.type,
-                'mime_type': stat_info.info.mime_type,
-                'idp': stat_info.info.owner.idp,
-                'permissions': stat_info.info.permission_set
+                'inode': {'storage_id': stat.info.id.storage_id,
+                          'opaque_id': stat.info.id.opaque_id},
+                'filepath': stat.info.path,
+                'userid': stat.info.owner.opaque_id,
+                'size': stat.info.size,
+                'mtime': stat.info.mtime.seconds,
+                'type': stat.info.type,
+                'mime_type': stat.info.mime_type,
+                'idp': stat.info.owner.idp,
+                'permissions': stat.info.permission_set
             }
-        elif stat_info.status.code == cs3code.CODE_NOT_FOUND:
-            self.log.info('msg="Failed stat" fileid="%s" reason="%s"' % (file_id, stat_info.status.message))
-            raise FileNotFoundError(stat_info.status.message + ", file " + file_id)
+        elif stat.status.code == cs3code.CODE_NOT_FOUND:
+            self.log.info('msg="Failed stat" fileid="%s" reason="%s"' % (file_path, stat.status.message))
+            raise FileNotFoundError(stat.status.message + ", file " + file_path)
         else:
-            self._handle_error(stat_info)
-
+            self._handle_error(stat)
+        
     def read_file(self, file_path, endpoint=None):
         """
         Read a file using the given userid as access token.
         """
         time_start = time.time()
-        
-        if self.storage_logic.stat(file_path, endpoint) is not None:
+
+        stat = self.storage_api.stat(file_path, endpoint)
+        if stat.status.code == cs3code.CODE_OK:
```

Comment:
Why don't we bail if code not ok? Otherwise, you will try to download a file whose stat failed.